go_modules/spec/dependabot/go_modules/dependency_grapher_spec.rb

```diff
@@ -112,34 +112,34 @@
             {
               name: "github.com/fatih/color",
               depends_on: [
-                "github.com/mattn/go-colorable",
-                "github.com/mattn/go-isatty",
-                "golang.org/x/sys"
+                "pkg:golang/github.com/mattn/go-colorable@v0.1.14",
+                "pkg:golang/github.com/mattn/go-isatty@v0.0.20",
+                "pkg:golang/golang.org/x/sys@v0.36.0"
               ]
             },
             {
               name: "github.com/mattn/go-colorable",
```

Comment:
I _think_ in order to properly match the relationship between this as a child of `github.com/fatih/color` and link up the transitive paths, we will need to use the PURL as the name/key as well, e.g.:

```diff
- github.com/mattn/go-colorable
+ pkg:golang/github.com/mattn/go-colorable@v0.1.14
```

If we don't do this I suspect we may not properly render the snapshot in the UI 🤔 

Since this means changing the key of the dependency collection, it might be worth using an Experiment to avoid modifying things for people who are staffshipped?

I don't think it's a huge deal, at worst they might temporarily lose transitive paths until they refresh/push again but changing the key makes me slightly more nervous than just changing `depends_on` as we have so far.